cmd/kubeadm/app/util/apiclient/dryrun.go

```diff
@@ -492,6 +510,47 @@ func (d *DryRun) DeleteBootstrapTokenReactor() *testing.SimpleReactor {
 	}
 }
 
+// ListPodsReactor returns a reactor that handles the LIST action on pods.
+func (d *DryRun) ListPodsReactor(nodeName string) *testing.SimpleReactor {
+	return &testing.SimpleReactor{
+		Verb:     "list",
+		Resource: "pods",
+		Reaction: func(action testing.Action) (bool, runtime.Object, error) {
+			a := action.(testing.ListAction)
+			if a.GetNamespace() != metav1.NamespaceSystem {
+				return false, nil, nil
+			}
+			restrictions := a.GetListRestrictions()
+			for _, res := range restrictions.Fields.Requirements() {
+				if res.Field == "spec.nodeName" {
+					nodeName = res.Value
```

Comment:
The `nodeName` seems to be duplicated with the `nodeName` parameter. 
Can we remove the input `nodeName` parameter here?